daemon/cluster/noderunner.go

```diff
@@ -0,0 +1,296 @@
+package cluster
+
+import (
+	"context"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	types "github.com/docker/docker/api/types/swarm"
+	"github.com/docker/docker/daemon/cluster/executor/container"
+	swarmapi "github.com/docker/swarmkit/api"
+	swarmnode "github.com/docker/swarmkit/node"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+)
+
+// nodeRunner implements a manager for contiously running swarmkit node, restarting them with backoff delays if needed.
+type nodeRunner struct {
+	nodeState
```

Comment:
How would you get the snapshot then? We would need to define the fields that can be copied twice and make a clone function to convert between them?